bastionzero/bzero
# CWC-1571: Agent sends its own heartbeats and ignores incoming ones

## Description of the change

Corresponding PRs in [bzero](https://github.com/bastionzero/bzero/pull/52) and [webshell-backend](https://github.com/bastionzero/webshell-backend/pull/1061)

  - Instead of waiting to receive regular healthcheck messages from Bastion and sending a response, Agent sends its own "I'm still alive" message every 20 seconds
  - Bastion processes these messages as normal
  - For backwards compatibility, Bastion continues sending healthcheck messages to agents older than 4.2.0. Newer agents receiving such a message log it as unrecognized. When all users have moved to the new version, this can be removed
  - Since Agent no longer receives an updated `targetName` from Bastion every heartbeat, remove `targetName` from its vault, and from the challenge request. It is only used at creation time as a part of registration

## To test

  - Create an agent
  - checkout `feat/push-heartbeats` in bzero and webshell-backend
  - Modify `agent.go`->`getAgentVersion` in agent.go to return "4.2.0" or greater
    - It will work even if you don't do this, but Bastion will continue sending the messages
  - rsync and restart the code for bzero and webshell-backend. 
  - You will see in the logs that Agent is sending healthcheck messages and Bastion is receiving them.

## Related JIRA tickets

Relates to JIRA: [CWC-1571](https://commonwealthcrypto.atlassian.net/browse/CWC-1571)

## Have you considered the security impacts?

Does this PR have any security impact?
